chips/nrf52840/src/ieee802154_radio.rs

```diff
@@ -801,7 +801,14 @@ impl<'a> Radio<'a> {
                     // or some instance in  driver taking buffer without properly replacing).
                     let rbuf = self.rx_buf.take().unwrap();
 
-                    let frame_len = rbuf[MIMIC_PSDU_OFFSET as usize] as usize - radio::MFR_SIZE;
+                    // data buffer format: | PSDU OFFSET | FRAME | LQI |
+                    // length of received data transferred to buffer (including PSDU)
+                    let data_len = rbuf[MIMIC_PSDU_OFFSET as usize] as usize;
+
+                    // lqi is found just after the data received
+                    let lqi = rbuf[data_len];
```

Comment:
To check that I understand: for the nRF52840 this value was already there in every packet we read, we were just never parsing it out before?

Reply:
That is mostly correct. The radio attempts to place the lqi into the buffer. However, we did not provide a buffer large enough for this value to be placed so it was not parsed and not in the `rbuf` struct.